src/scripts/main.js

```diff
@@ -1,3 +1,199 @@
+const cellsInRow = rows.map((row) => [...row.querySelectorAll('.field-cell')]);
+const cellStartValues = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+for (let i = 0; i < 4; i++) {
+  for (let k = 0; k < 4; k++) {
+    columns[k].push(rows[i].children[k]);
+  }
+}
+
+let clearCells = [...cells];
+let started = false;
+let rotated = false;
+
+function randomCellFill() {
+  if (!clearCells.length) {
+    return;
+  }
+
+  const randomIndex = (someArr) => Math.floor(Math.random() * someArr.length);
+  const randomCell = clearCells[randomIndex(clearCells)];
+  const randomValue = cellStartValues[randomIndex(cellStartValues)];
+
+  randomCell.classList = (`field-cell field-cell--${randomValue}`);
+  randomCell.innerText = randomValue;
+
+  clearCells.splice(clearCells.indexOf(randomCell), 1);
+}
+
+const start = () => {
+  startMessage.style = 'display: none;';
+  startButton.classList = 'button restart';
+  startButton.innerText = 'Restart';
+
+  randomCellFill();
+  randomCellFill();
+  started = true;
+};
+
+const restart = () => {
+  loseResult.classList.add('hidden');
+  winResult.classList.add('hidden');
+  clearCells = [...cells];
+  gameScore.innerText = 0;
+
+  cells.forEach(cell => {
+    cell.innerText = '';
+    cell.className = 'field-cell';
+  });
+
+  randomCellFill();
+  randomCellFill();
+};
+
+function isMergePossible() {
+  const wholeField = [...cellsInRow, ...columns];
+
+  for (const line of wholeField) {
+    for (let i = 0; i < 3; i++) {
+      if (line[i].innerText === line[i + 1].innerText) {
+        return true;
+      }
+    }
+  }
+}
+
+function cellUp() {
+  for (const column of columns) {
+    rotateCells([...column].reverse());
+  }
+}
+
+function cellDown() {
+  for (const column of columns) {
+    rotateCells(column);
+  }
+}
+
+function cellRight() {
+  for (const row of cellsInRow) {
+    rotateCells(row);
+  }
+}
+
+function cellLeft() {
+  for (const row of cellsInRow) {
+    rotateCells([...row].reverse());
+  }
+}
+
+function arrowMove(direction) {
+  if (!clearCells.length && !isMergePossible()) {
+    loseResult.classList.remove('hidden');
+  }
+
+  switch (direction) {
+    case 'ArrowUp':
+      cellUp();
+      break;
+    case 'ArrowDown':
+      cellDown();
+      break;
+    case 'ArrowRight':
+      cellRight();
+      break;
+    case 'ArrowLeft':
+      cellLeft();
+      break;
+  }
+
+  cells.forEach(cell => {
+    cell.removeAttribute('data-blocked');
+  });
+}
+
+function deleteElement(element) {
+  element.innerText = '';
+  element.className = 'field-cell';
+  clearCells.push(element);
+}
+
+function rotateElement(curr, prev) {
+  curr.innerText = prev.innerText;
+  curr.className = prev.className;
+  clearCells.splice(clearCells.indexOf(curr), 1);
+  deleteElement(prev);
+}
+
+function cellMerge(curr, prev) {
+  const value = curr.innerText * 2;
+
+  curr.innerText = value;
+  curr.className = `field-cell field-cell--${value}`;
+  gameScore.innerText = +gameScore.innerText + value;
```

Comment:
use `Number` instead `+`
```suggestion
  gameScore.innerText = Number(gameScore.innerText) + value;
```